src/main/java/it/gov/pagopa/payhub/pdnd/service/PdndService.java

```diff
@@ -0,0 +1,46 @@
+package it.gov.pagopa.payhub.pdnd.service;
+
+import com.nimbusds.jose.JOSEException;
+import it.gov.pagopa.payhub.pdnd.connector.pdnd.client.PdndClientImpl;
+import it.gov.pagopa.payhub.pdnd.connector.pdnd.service.PdndClientAssertionBuilderService;
+import it.gov.pagopa.payhub.pdnd.exception.custom.JwtClaimBuildException;
+import it.gov.pagopa.payhub.pdnd.model.PdndGenericConfig;
+import it.gov.pagopa.payhub.pdnd.utils.JWTUtils;
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class PdndService {
+
+  private final PdndClientImpl pdndClientImpl;
+  private final PdndClientAssertionBuilderService pdndClientAssertionBuilderService;
+  protected final ConcurrentHashMap<PdndGenericConfig, String> jwtCache = new ConcurrentHashMap<>();
+
+  public PdndService(PdndClientImpl pdndClientImpl,
+      PdndClientAssertionBuilderService pdndClientAssertionBuilderService) {
+    this.pdndClientImpl = pdndClientImpl;
+    this.pdndClientAssertionBuilderService = pdndClientAssertionBuilderService;
+  }
+
+  public String generateToken(PdndGenericConfig pdndGenericConfig) {
+    return jwtCache.compute(pdndGenericConfig, (key, existingJwt) -> {
+      log.debug("Check cache for token exists and not expired");
```

Comment:
let's print also the PdnGenericConfig class name, in order to know about the configuration that is going to be configured